### PR TITLE
Simplified MHV Landing Page for unregistered and unverified users

### PR DESCRIPTION
**Pull request by carlosfelix2**
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary
We reviewed what content is shown to the different types of users and made changes to simplify the page for unverified and unregistered users.
- Unverified users do not see the resource links (hub links), helpdesk info, "learn more" expandable area in the header,  or newsletter signup
- Unregistered users (not a VA patient) do not see the resource links (hub links), helpdesk info, "learn more" expandable area in the header, or newsletter signup
- Users without an MHV account see all components including the appropriate alert.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/85897

## Testing done
- Updated unit and e2e tests

## Screenshots
|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |   localhost_3001_my-health_(iPhone 14 Pro Max) (5)     |  localhost_3001_my-health_(iPhone 14 Pro Max) (5)  |
| Desktop |    localhost_3001_my-health_ (10)    |   localhost_3001_my-health_ (10)  |

## What areas of the site does it impact?
- MHV Landing page

## Acceptance criteria
- Implemented the hiding of some landing page content per designs when the "no access to MHV" alert is displayed
  - secondary nav
  - expanding alert
  - links below "need help?"

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [X] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [X] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [X] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [X] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

File: src/applications/mhv-landing-page/components/HeaderLayout.jsx [hunk 1]
```diff
@@ -83,52 +86,54 @@ const HeaderLayout = ({ showWelcomeMessage = false }) => {
...
-                      data-testid="mhv-go-back-2"
-                      href={goBackUrl}
-                    >
-                      {goBackLinkText}
-                    </a>
-                  </p>
-                </div>
-              </va-alert-expandable>
-            </div>
+            {showLearnMore && (
```
Comment (carlosfelix2) [inline src/applications/mhv-landing-page/components/HeaderLayout.jsx:89]: Note: Only the `showLearnMore` boolean flag was added. All the other changes here are indentation.

File: src/applications/mhv-landing-page/components/HeaderLayout.jsx [hunk 2]
```diff
@@ -147,20 +152,23 @@ const HeaderLayout = ({ showWelcomeMessage = false }) => {
...
-      <div
-        className={classnames(
-          'vads-u-border-color--gray-light',
-          'vads-u-border-bottom--1px',
-          'vads-u-margin-bottom--3',
-        )}
-      >
-        {showWelcomeMessage && <WelcomeContainer />}
-      </div>
+      {showWelcomeMessage && (
```
Comment (carlosfelix2) [inline src/applications/mhv-landing-page/components/HeaderLayout.jsx:155]: Note: This moves the line to be part of the welcome message if it is shown.

File: src/applications/mhv-landing-page/components/HubLinks.jsx [hunk 3]
```diff
@@ -41,7 +41,7 @@ const HubLinks = ({ hubs }) => {
   ));
   return (
     <div className="vads-l-grid-container large-screen:vads-u-padding-x--0">
-      <div className="vads-l-row vads-u-margin-bottom--3">{hubLayout}</div>
+      <div className="vads-l-row">{hubLayout}</div>
```
Comment (carlosfelix2) [inline src/applications/mhv-landing-page/components/HubLinks.jsx:44]: Note: This removes extra padding that was not needed.

**Review by dcloud — APPROVED** [review]
This looks good to me. Checked for unverified and unregistered users.

**Pull request merged**